### ISSUE TITLE
Performance/stuttering issue when opening and closing notes

### ISSUE THREAD

**Issue by alexlyzhov** [isaaclyman/novel-word-count-obsidian#18]
Hi, thanks for the plugin, it's amazing!
I had this stuttering in Obsidian happening when I'm opening and closing notes that was driving me mad. I looked for the cause using https://github.com/chrisgrieser/obsidian-divide-and-conquer binary search and it's novel word count. It's happening even when all other plugins except novel word count are disabled.

Every time I open or close a note, the plugin is likely doing something expensive because it stutters quite a bit. It doesn't happen immediately after I do the action but rather with some sub-second delay.

MacOS 12 / MacBook Pro'21 / latest versions of Obsidian and plugin.

Debug output in the console looks like this every time I open or close any note in Obsidian:
```
novel-word-count: [layout-change] hook fired recounting all files
novel-word-count: refreshAllCounts
novel-word-count|getAllFileCounts: 165.601806640625 ms
novel-word-count: Setting display counts for 7520 fileItems
novel-word-count|updateDisplayedCounts: 778.837890625 ms
```

**Comment by alexlyzhov** [comment]
Possibly this change doesn't do what it's intended to do https://github.com/isaaclyman/novel-word-count-obsidian/commit/5cfa3a8a61a510fee9b46f2701c9e2b4f94e1529#diff-564c860307257bfac9176f554a35aa858c350a59d75c0dcd0a5d63a480805bd0R303

**Comment by isaaclyman** [comment]
Hmmm...yeah, that layout-change hook fires more often than I'd like. I'll look into it.

**Comment by isaaclyman** [comment]
Fixed in 2.9.1. I've added a check to make sure we update the display when you change workspaces, but not when you open/close a file.

Referencing #9 for my own notes.

**Comment by isaaclyman** [comment]
@alexlyzhov 7,500 notes (as your debug output states) is incredible, by the way. What a vault. Let me know if you see any other performance issues, as I don't have a vault of nearly that size to test on.
